### ISSUE TITLE
Support for reader conditionals

### ISSUE THREAD

**Issue by venantius** [weavejester/cljfmt#59]
Currently, cljfmt formats reader conditionals in the same way code is otherwise formatted. Unfortunately, this leads to code that looks like the following:
```clojure
#? (:clj (throw (IllegalArgumentException.
                  "reason"))
          :cljs (throw "reason")))
```
This is in contrast to what we would expect, based on the examples for good code given in http://clojure.org/guides/reader_conditionals, which would suggest code formatting like the following:
```clojure
#?(:clj (throw (IllegalArgumentException.
                  "reason"))
   :cljs (throw "reason")))
```

Note the two issues here:
 * a space after the reader conditional macro `#?` that shouldn't be there, and
 * incorrect indentation of keywords within the macro

**Comment by rjohnsondev** [comment]
Adding to this, the current behaviour can produce broken code when using the splicing reader conditional:

```
(:require [clojure.string :as s]
           #?@(:clj  [[clj-time.format :as tf]
                      [clj-time.coerce :as tc]]
               :cljs [[cljs-time.coerce :as tc]
                      [cljs-time.format :as tf]])
```

becomes

```
    #? @(:clj  [[clj-time.format :as tf]
                [clj-time.coerce :as tc]]
               :cljs [[cljs-time.coerce :as tc]
                      [cljs-time.format :as tf]]
```

which will result in the exception: `CompilerException java.lang.RuntimeException: read-cond body must be a list` due to the extra space before the `@` in the conditional.

**Comment by weavejester** [comment]
This issue relies on reader conditional support being added to [rewrite-clj](https://github.com/xsc/rewrite-clj) and [rewrite-cljs](https://github.com/rundis/rewrite-cljs).

**Comment by carrete** [comment]
> This issue relies on reader conditional support being added to rewrite-clj and rewrite-cljs.

Just added to rewrite-clj, https://github.com/xsc/rewrite-clj/blob/master/CHANGES.md#052

**Comment by carrete** [comment]
And now rewrite-cljs, https://github.com/rundis/rewrite-cljs/issues/8

**Comment by weavejester** [comment]
Should be fixed by #80.
